openlibrary/templates/books/custom_carousel.html

```diff
@@ -15,6 +15,8 @@
   $if book.get('cover_url'):
     $ cover_url = book.get('cover_url')
   $elif book.get('cover_id') or book.get('cover_i'):
```

Comment:
This condition is designed to only be satisfied if a value for 'cover_id' is found i.e. if `book.get('cover_id')` returns a value that evaluates to **TRUE**.

However, for some books that don't have a cover ID, `book.get('cover_id')` is returning the string **'undefined'**. And as 'undefined' technically evaluates to **TRUE**, the code inside this condition is running, even though no valid cover id has been provided.

I believe that's the root cause of this issue: #6648

My proposed solution is here: #6722